### DISCUSSION TITLE
max header size applied only on HTTP/1

### DISCUSSION THREAD

**Discussion by oussama-zaoui** [quarkusio/quarkus#35020] [Q&A]
Hello everyone,

I am currently encountering a peculiar issue with the property "quarkus.http.limits.max-header-size=30k" in my Quarkus application. Oddly, this property seems to work perfectly when I am in HTTP mode. However, as soon as I switch to HTTPS, the max header size is drastically reduced to around 8k, causing 431 Request Header Fields Too Large.

To address this problem and restore the expected behavior with the larger header size, I found a temporary solution by adding another property, "quarkus.http.http2=false," to disable HTTP2.

Quarkus Version Used: 2.12.1.Final

I am considering opening a discussion in the Quarkus repository to seek insights and potential solutions for this issue. Any suggestions or help would be greatly appreciated. Thank you all for your time and support in advance.

Best regards,

**Comment by yashumitsu** [comment]
There is a recent change: #33692

**Reply by codespearhead** [reply]
That issue is now closed.

@oussama-zaoui should this discussion be closed as well?
